In [3]:
import pandas as pd
import joblib
from sklearn.metrics import accuracy_score, precision_score, recall_score, f1_score, classification_report
from sklearn.model_selection import cross_val_score, StratifiedKFold

# === 1. Wczytanie modelu i danych walidacyjnych ===
model = joblib.load("best_model_rf.pkl")
df_val = pd.read_csv("validation_set.csv")

# Zmienna docelowa (zmień 'target' jeśli Twoja kolumna ma inną nazwę)
target_column = 'target'
X_val = df_val.drop(columns=[target_column])
y_val = df_val[target_column]

# === 2. Predykcja na zbiorze walidacyjnym ===
y_pred = model.predict(X_val)

# Metryki na zbiorze walidacyjnym
acc = accuracy_score(y_val, y_pred)
prec = precision_score(y_val, y_pred, average='weighted', zero_division=0)
rec = recall_score(y_val, y_pred, average='weighted')
f1 = f1_score(y_val, y_pred, average='weighted')

print("Metryki na zbiorze walidacyjnym:")
print(f"Dokładność (Accuracy): {acc:.4f}")
print(f"Precyzja (Precision):  {prec:.4f}")
print(f"Czułość (Recall):      {rec:.4f}")
print(f"F1-score:              {f1:.4f}")
print("\nClassification report:")
print(classification_report(y_val, y_pred, zero_division=0))

# === 3. Walidacja krzyżowa (cross-validation) ===
cv = StratifiedKFold(n_splits=5, shuffle=True, random_state=42)

# F1-score z CV
cv_f1_scores = cross_val_score(model, X_val, y_val, cv=cv, scoring='f1_weighted')
# Accuracy z CV
cv_acc_scores = cross_val_score(model, X_val, y_val, cv=cv, scoring='accuracy')

print("\nWalidacja krzyżowa (5-krotna):")
print(f"Średni F1-score (CV):     {cv_f1_scores.mean():.4f}")
print(f"Średnia dokładność (CV):  {cv_acc_scores.mean():.4f}")



/usr/lib/python3/dist-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.2 when using version 1.4.1.post1. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:
https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations
  warnings.warn(
/usr/lib/python3/dist-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.2 when using version 1.4.1.post1. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:
https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations
  warnings.warn(


Metryki na zbiorze walidacyjnym:
Dokładność (Accuracy): 0.6057
Precyzja (Precision):  0.6369
Czułość (Recall):      0.6057
F1-score:              0.6171

Classification report:
              precision    recall  f1-score   support

           0       0.75      0.65      0.70     57589
           1       0.38      0.49      0.43     25165

    accuracy                           0.61     82754
   macro avg       0.57      0.57      0.56     82754
weighted avg       0.64      0.61      0.62     82754


Walidacja krzyżowa (5-krotna):
Średni F1-score (CV):     0.5964
Średnia dokładność (CV):  0.5804


Metryka             | Oryginalna grupa  | Nasza walidacja
F1-score (Test)     | 0.4427            | 0.6171
Dokładność (Test)   | 0.5804            | 0.6057
Precyzja (Test)     | 0.3725            | 0.6369
Czułość (Test)      | 0.5454            | 0.6057
F1-score (CV)       | 0.4382            | 0.5964
Dokładność (CV)     | 0.5936            | 0.5804